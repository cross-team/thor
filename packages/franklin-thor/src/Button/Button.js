import React from 'react';
import PropTypes from 'prop-types';
// import Styled from './styles';

function Button({ children, variant, accent, onClick, ...props }) {
  let color = '#39adb5'; /* theme.primary */
  if (accent) {
    if (accent === 'secondary' || accent === 'primary') {
      color = '#e53935'; /* theme[accent] */
    } else {
      color = accent;
    }
  }
  return (
    <button type={'button'} variant={variant} accent={color} onClick={onClick} {...props}>
      <span>{children}</span>
    </button>
  );
}

Button.propTypes = {
  accent: PropTypes.string,
  children: PropTypes.element,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  variant: PropTypes.oneOf(['filled', 'outlined', 'link']),
};

Button.defaultProps = {
  accent: 'primary',
  children: undefined,
  disabled: false,
  variant: 'filled',
  onClick() {
    // eslint-disable-next-line no-console
    console.log('Button clicked...');
  },
};

export default Button;
