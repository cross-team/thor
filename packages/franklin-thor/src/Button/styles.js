// import styled from 'styled-components';

// const styles = {
//   StyledButtom: styled.button`
//     background-color: ${({ accent, variant }) => (variant === 'filled' ? accent : '')};
//     border: ${({ accent, variant }) =>
//       `1px solid ${variant === 'outlined' ? accent : 'transparent'}`};
//     border-radius: 3px !important;
//     color: ${props => props.color};
//     cursor: pointer;
//     display: inline-block;
//     padding: 6px 12px 7px 12px;
//     text-align: center;
//     text-decoration: none;
//     text-shadow: none;
//     white-space: nowrap;
//     &:hover {
//       text-decoration: ${({ variant }) => (variant === 'link' ? 'underline' : '')};
//       transform: scale(1.05);
//     }
//     &:focus {
//       outline-width: 0;
//     }
//   `,
//   ButtonContent: styled.span`
//     display: flex;
//   `,
// };

// export default styles;
