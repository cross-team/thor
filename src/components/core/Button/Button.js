import React from 'react';
import PropTypes from 'prop-types';
import Styled from './styles';

function Button({ children, variant, accent, ...props }) {
  let color = '#0000ff'; /*theme.primary*/
  if (accent) {
    if (accent === 'secondary' || accent === 'primary') {
      color = '#ff0000'; /*theme[accent]*/
    } else {
      color = accent;
    }
  }
  return (
    <Styled.StyledButtom variant={variant} accent={color} {...props}>
      <Styled.ButtonContent>{children}</Styled.ButtonContent>
    </Styled.StyledButtom>
  );
}

Button.propTypes = {
  children: PropTypes.any,
  variant: PropTypes.oneOf(['filled', 'outlined', 'link']),
  accent: PropTypes.string
};

Button.defaultProps = {
  variant: 'filled'
};

export default Button;
