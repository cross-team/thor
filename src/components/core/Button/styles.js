import styled from 'styled-components';

const styles = {
  StyledButtom: styled.button`
    background: 0;
    background-color: ${({ accent, variant }) =>
      variant === 'filled' ? accent : ''};
    border: ${({ accent, variant }) =>
      `1px solid ${variant === 'outlined' ? accent : 'transparent'}`};
    border-radius: 3px !important;
    color: #fff !important;
    cursor: pointer;
    display: inline-block;
    font-size: 13px !important;
    font-weight: bold;
    line-height: 17px;
    margin: 1px;
    padding: 6px 12px 7px 12px;
    position: relative;
    text-align: center;
    text-decoration: none;
    text-shadow: none;
    white-space: nowrap;
    transition-duration: 0.1s;
    transition-property: transform;
    &:hover {
      text-decoration: ${({ variant }) =>
        variant === 'link' ? 'underline' : ''};
      transform: scale(1.05);
    }
    &:active {
      border-color: #202020 !important;
    }
    &:focus {
      outline-width: 0;
    }
  `,
  ButtonContent: styled.span`
    display: flex;
  `
};

export default styles;
