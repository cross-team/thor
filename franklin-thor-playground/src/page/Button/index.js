import React from 'react'
import { storiesOf } from '@storybook/react'
// import { boolean, text, select } from '@storybook/addon-knobs';
import Button from '@cross.team/core/Button'
import defaultTheme from '../../themes/default'
import { ThemeProvider } from '@material-ui/styles'

const withTheme = story => <ThemeProvider theme={defaultTheme}>{story()}</ThemeProvider>

storiesOf('Button', module)
  .addDecorator(withTheme)
  .add('with text', () => <Button> Hi Button</Button>)
  .add('with emoji', () => (
    <Button>
      <span role="img" aria-label="so cool">
        😀 😎 👍 💯
      </span>
    </Button>
  ))
