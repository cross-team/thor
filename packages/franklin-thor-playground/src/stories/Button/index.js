import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean, text, select } from '@storybook/addon-knobs';
import Button from '@franklin-thor/core';

console.log('Aquiii!!!!!!!');
console.log(JSON.stringify(Button, null, 2));

const label = 'Variant';
const options = {
  Filled: 'filled',
  Outlined: 'outlined',
  Link: 'link',
};
const defaultValue = 'filled';
const groupId = 'Default';

storiesOf('Button', module)
  .add('with text', () => (
    <Button
      accent={text('Accent', 'primary', groupId)}
      disabled={boolean('Disabled', false, groupId)}
      variant={select(label, options, defaultValue, groupId)}
    >
      Hello Button
    </Button>
  ))
  .add('with emoji', () => (
    <Button accent="primary">
      <span role="img" aria-label="so cool">
        😀 😎 👍 💯
      </span>
    </Button>
  ));
