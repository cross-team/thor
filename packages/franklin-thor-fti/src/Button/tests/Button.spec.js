import Button from '../Button'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'

configure({ adapter: new Adapter() })

let component
beforeEach(() => {
  component = shallow(<Button variant="outlined" id="btn-test" />)
})

describe('<Button /> rendering', () => {
  it('should render one <Button>', () => {
    expect(component.find('#btn-test')).toHaveLength(1)
    expect(component).toMatchSnapshot()
  })
})
