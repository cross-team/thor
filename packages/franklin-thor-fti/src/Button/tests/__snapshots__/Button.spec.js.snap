// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> rendering should render one <Button> 1`] = `ShallowWrapper {}`;
